main.js

```diff
@@ -57,26 +57,35 @@ document.addEventListener('DOMContentLoaded', function(e) {
     };
 
     const doDoh = function() {
-        document.getElementById('doh-url-form').classList.add('was-validated');
-
         responseElem.childNodes.forEach(node => node.remove());
+        $loadingModal.modal('show');
+        document.getElementById('do-doh').disabled = true;
 
-        let url = urlInputElem.value;
-        if (!url) {
-            return;
-        }
+        let url = urlInputElem.value || "https://cloudflare-dns.com/dns-query";
         if (corsSwitch.checked) {
             url = cors_proxy + url;
         }
         const method = document.getElementById('doh-method').value || 'POST';
         const qname = document.getElementById('doh-qname').value || '.';
         const qtype = document.getElementById('doh-qtype').value || 'A';
-        $loadingModal.modal('show');
-        document.getElementById('do-doh').disabled = true;
-        const resolver = new doh.DohResolver(url);
-        resolver.query(qname, qtype, method)
-          .then(successFunction)
-          .catch(errorFunction);
+
+        const query = doh.makeQuery(qname, qtype);
+        if (dnssecSwitch.checked) {
+            query.additionals =  [{
+                type: 'OPT',
+                name: '.',
+                udpPayloadSize: 4096,
+                flags: 1 << 15, // DO bit
```

Comment:
@kimbo I tried using `doh.DNSSEC_OK` here (which my IDE even suggested) but it was undefined when testing.

Reply:
We'll have to expose `dnsPacket` in dohjs, which will make things a lot more flexible, actually.